src/main/java/faang/school/projectservice/validator/InternshipValidator.java

```diff
@@ -0,0 +1,70 @@
+package faang.school.projectservice.validator;
+
+import faang.school.projectservice.dto.client.InternshipToCreateDto;
+import faang.school.projectservice.dto.client.InternshipToUpdateDto;
+import faang.school.projectservice.exception.DataValidationException;
+import faang.school.projectservice.mapper.InternshipMapper;
+import faang.school.projectservice.model.Internship;
+import faang.school.projectservice.model.TeamMember;
+import faang.school.projectservice.repository.InternshipRepository;
+import faang.school.projectservice.repository.ProjectRepository;
+import faang.school.projectservice.repository.TeamMemberRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static faang.school.projectservice.exception.InternshipError.*;
+
+@Component
+@RequiredArgsConstructor
+public class InternshipValidator {
+    private final InternshipRepository internshipRepository;
+    private final ProjectRepository projectRepository;
+    private final TeamMemberRepository teamMemberRepository;
+    private final InternshipMapper internshipMapper;
```

Comment:
Точно ли нужен данный маппер?
```suggestion
    private final InternshipMapper internshipMapper;
```

Reply:
точно, забыла убрать)